### PR TITLE
feat: Implement minimum password check on client and server

### LINKED ISSUE #590: Implement Minimum Password Length Check on Update Password Page
### Guidelines 👇🏼
If you would like to be rewarded by BoxyHQ for your participation in contributing to Hacktoberfest, [please register for our Hacktoberfest event](https://events.mlh.io/events/10461-boxyhq-hacktoberfest-2023).

___

**Description:**
Currently, the password update page (/settings/password) allows users to set passwords of any length. To enhance security and ensure strong password practices, we should add a minimum password length check on both the client and server sides.

**Expected Behavior:**
[NIST](https://pages.nist.gov/800-63-3/sp800-63-3.html) recommends a password length of at least 8.
When users attempt to update their password on the **/settings/password** page, the system should validate that the new password meets the minimum length requirement. This validation should occur in real time on the client side for immediate feedback and on the server side to enforce security standards.

**Proposed Solution:**
- Implement client-side validation that checks the minimum password length as users type.
- Enforce server-side validation to reject password updates that don't meet the minimum length requirement.
- Provide clear error messages for users attempting to set a password that's too short.

**Additional Information:**
Implementing this validation will enhance the security of user accounts by ensuring that passwords meet a minimum length requirement, a common best practice in password management.

### PR DESCRIPTION
**Pull request by Dhoni77**
Closes #590 

- Implemented min 8 character length validation on client
- Implemented password validation and error handling on server side.

### CODE REVIEW

File: components/account/UpdatePassword.tsx [hunk 1]
```diff
@@ -8,8 +8,8 @@ import { Card, InputWithLabel } from '@/components/shared';
 import { defaultHeaders } from '@/lib/common';
 
 const schema = Yup.object().shape({
-  currentPassword: Yup.string().required(),
-  newPassword: Yup.string().required().min(7),
+  currentPassword: Yup.string().required().min(8),
```
Comment (devkiran) [inline components/account/UpdatePassword.tsx:11]:
```suggestion
  currentPassword: Yup.string().required(),
```
We can't force min length check on current password.

**Review by devkiran — CHANGES_REQUESTED** [review]
Can you please revert all changes related to checking password complexity (Keep the length check). We'll create an issue to discuss this.

**Comment by Dhoni77** [comment]
> Can you please revert all changes related to checking password complexity (Keep the length check). We'll create an issue to discuss this.

Ok Sure

**Comment by Dhoni77** [comment]
cc @devkiran done the requested changes

File: pages/auth/login.tsx [hunk 2]
```diff
@@ -64,7 +64,7 @@ const Login: NextPageWithLayout<
     },
     validationSchema: Yup.object().shape({
       email: Yup.string().required().email(),
-      password: Yup.string().required(),
+      password: Yup.string().required().min(8),
```
Comment (devkiran) [inline pages/auth/login.tsx:67]: @Dhoni77 Please don't do min password length check during login. It's not a good practice.

**Review by devkiran — CHANGES_REQUESTED** [review]

File: pages/auth/login.tsx (hunk 2 above)
Comment (Dhoni77) [reply pages/auth/login.tsx]: done

**Review by devkiran — APPROVED** [review]

**Pull request merged**